crates/bevy_ecs/src/world/mod.rs

```diff
@@ -46,6 +50,28 @@ pub struct World {
     /// Access cache used by [WorldCell].
     pub(crate) archetype_component_access: ArchetypeComponentAccess,
     main_thread_validator: MainThreadValidator,
+    pub(crate) global_system_counter: AtomicU32,
```

Comment:
In my mind this isn't a "system counter". Its more like a "change context tick". World could be used without systems and `global_system_counter` would still have meaning. We might later choose to store a "change context tick" in QueryState in addition to (or instead of) systems and it would be equally valid. It also isn't a "change context counter". It doesn't count the number systems/change contexts. It counts the number of times a context is run. I think tick is a decent fit for that (and is relatively concise).

I would like to propose the following renames:

* `global_system_counter` -> `change_tick` (the "global" is implicit)
* `exclusive_system_counter` -> `last_change_tick` (this is the "last change tick" from the raw World's perspective. exclusive systems override this value, but that is a layer "on top" of World, not something World needs to know about)
* `system_counter` (from the context of a specific system) -> `last_change_tick`

Reply:
I agree with these renames.